crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -0,0 +1,203 @@
+pub mod common;
+
+// Important! Use unique file names for dump files, tests can be run in parallel.
+mod dump_and_load_tests {
+    use std::process::Command;
+
+    use starknet_rs_providers::Provider;
+
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::remove_file;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    use std::sync::Arc;
+
+    use starknet_rs_accounts::{Account, ExecutionEncoding, SingleOwnerAccount};
+    use starknet_rs_contract::ContractFactory;
+    use starknet_rs_core::chain_id;
+    use starknet_rs_core::types::{BlockId, BlockTag, FieldElement};
+
+    use crate::common::utils::{
+        get_events_contract_in_sierra_and_compiled_class_hash, get_predeployed_account_props,
+    };
+
+    #[tokio::test]
+    async fn check_dump_path_with_dump_on() {
+        let devnet_dump =
+            BackgroundDevnet::spawn_with_additional_args(Some(["--dump-on", "exit"].to_vec()))
+                .await;
+        assert!(devnet_dump.is_err());
+    }
+
+    #[tokio::test]
+    async fn mint_dump_on_transaction_and_load() {
+        // dump after transaction
+        let dump_file_name = "dump_on_transaction";
+        let devnet_dump = BackgroundDevnet::spawn_with_additional_args(Some(
+            ["--dump-path", dump_file_name, "--dump-on", "transaction"].to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let mint_tx_hash = devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        // load transaction from file and check transaction hash
+        let devnet_load = BackgroundDevnet::spawn_with_additional_args(Some(
+            ["--dump-path", dump_file_name].to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let loaded_transaction =
+            devnet_load.json_rpc_client.get_transaction_by_hash(mint_tx_hash).await.unwrap();
+        if let starknet_rs_core::types::Transaction::Invoke(
+            starknet_rs_core::types::InvokeTransaction::V1(invoke_v1),
+        ) = loaded_transaction
+        {
+            assert_eq!(invoke_v1.transaction_hash, mint_tx_hash);
+        } else {
+            panic!("Could not unpack the transaction from {loaded_transaction:?}");
+        }
+
+        remove_file(dump_file_name);
+    }
+
+    #[tokio::test]
+    async fn mint_dump_on_exit_and_load() {
+        // dump after transaction
+        let dump_file_name = "dump_on_exit";
+        let devnet_dump = BackgroundDevnet::spawn_with_additional_args(Some(
+            ["--dump-path", dump_file_name, "--dump-on", "exit"].to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let devnet_dump_pid = devnet_dump.process.id();
+        let mint_tx_hash = devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        #[cfg(windows)]
+        {
+            // To send SIGINT signal on windows, windows-kill is needed
```

Comment:
Ok, but how does a user handle this? They can't press ctrl+c?

Reply:
They can, poblem here was to do it from code level in tests in windows.